### PR TITLE
Getting rid of tutorial parsing duplication

### PR DESCRIPTION
**Pull request by riknoll**
In v1 of the tutorial tool I just duplicated the tutorial parsing code because I didn't want to make changes to pxt. Now that it's shipped, I'm removing the duplication and adding an editorcontroller message that imports a tutorial from just the markdown.

This will temporarily break the tutorial tool for all targets until they get bumped. We should do a bump of mc with these changes ASAP once this is merged, so I'm putting the "DO NOT MERGE" label on it

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -3908,3 +3907,30 @@ document.addEventListener("DOMContentLoaded", () => {
         }
     }, false);
 })
+
+
+function getTutorialOptions(md: string, tutorialId: string, filename: string, reportId: string, recipe: boolean): [pxt.tutorial.TutorialOptions, string] {
```
Comment (pelikhan) [inline webapp/src/app.tsx]: can you return a map that explains what the return values mean? ```{ options: , ???: string }```

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -1462,10 +1462,29 @@ export class ProjectView
...
+
+            return this.createProjectAsync({
+                name: "untitled",
+                tutorial: options,
+                preferredEditor: editor,
+                dependencies
+            });
+        }
+        catch (e) {
+            Util.userError("Could not import tutorial");
```
Comment (pelikhan) [inline webapp/src/app.tsx]: add a tick?

Comment (riknoll) [reply webapp/src/app.tsx]: It doesn't swallow it, it rejects the promise. I'm not really sure that there is a good reason to pollute our exceptions with errors generated by the tutorial tool

**Comment by pelikhan** [comment]
Point to /beta?

**Comment by riknoll** [comment]
I have beta and non-beta endpoints for all of the targets in the dropdown. I can comment out the ones that don't work (all but minecraft) if that helps

**Comment by pelikhan** [comment]
Minecraft is the only user of the tool. We haven’t advertised it anywhere.

**Pull request merged**